### PR TITLE
DAOS-13328 test: Set control_metadata in server config

### PR DESCRIPTION
**Pull request by phender**
Set the control_metadata path in the server config when using metadata on ssd.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-nvme: auto_md_on_ssd

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [x] You are the appropriate gatekeeper to be landing the patch.
* [x] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [x] Githooks were used. If not, request that user install them and check copyright dates.
* [x] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [x] All builds have passed.  Check non-required builds for any new compiler warnings.
* [x] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [x] If applicable, the PR has addressed any potential version compatibility issues.
* [x] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [x] Extra checks if forced landing is requested
  * [x] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [x] No new NLT or valgrind warnings.  Check the classic view.
  * [x] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/launch.py [hunk 1]
```diff
@@ -1561,6 +1564,8 @@ def _add_auto_storage_yaml(self, storage_info, yaml_dir, tier_0_type, scm_size,
             scm_size (int): scm_size to use with ram storage tiers
             scm_mount (str): the base path for the storage tier 0 scm_mount.
             max_nvme_tiers (int): maximum number of NVMe tiers to generate
+            control_metadata (str, optional): directory to store control plane metadata when using
+                metadata on SSD. Defaults to None.
```
Comment (daltonbohning) [inline src/tests/ftest/launch.py:1568]: The caller defaults this to `None`, but not this function. Should it?

**Review by daltonbohning — APPROVED** [review]

**Review by tanabarr — APPROVED** [review]

**Comment by phender** [comment]
Test Failures:

- NEW issue https://daosio.atlassian.net/browse/DAOS-13375:
  - [Test Hardware / Functional Hardware Medium / FTEST_daos_test.DAOS_Rebuild.REBUILD0: drop rebuild scan reply](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/1/testReport/junit/FTEST_daos_test/DAOS_Rebuild/Test_Hardware___Functional_Hardware_Medium___REBUILD0__drop_rebuild_scan_reply/)

- Known issue https://daosio.atlassian.net/browse/DAOS-13219:
  - [Test Hardware / Functional Hardware Medium / FTEST_daos_vol.DaosVol.5-./daos_vol/h5_suite.py:DaosVol.test_daos_vol_mpich;run-container-daos_vol_tests-test5-hosts-job_manager-pool-server_config-engines-0-storage-0-1-2-fdd6](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/1/testReport/junit/FTEST_daos_vol/DaosVol/Test_Hardware___Functional_Hardware_Medium___5___daos_vol_h5_suite_py_DaosVol_test_daos_vol_mpich_run_container_daos_vol_tests_test5_hosts_job_manager_pool_server_config_engines_0_storage_0_1_2_fdd6/)

- Known issue https://daosio.atlassian.net/browse/DAOS-12750
  - [Test Hardware / Functional Hardware Large / FTEST_erasurecode.EcodIor.1-./erasurecode/ior_smoke.py:EcodIor.test_ec;run-container-hosts-ior-client_processes-iorflags-objectclass-transfersize_blocksize-1M_128M-pool-server_config-engines-0-storage-0-1-2-setup-53e1](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/1/testReport/junit/FTEST_erasurecode/EcodIor/Test_Hardware___Functional_Hardware_Large___1___erasurecode_ior_smoke_py_EcodIor_test_ec_run_container_hosts_ior_client_processes_iorflags_objectclass_transfersize_blocksize_1M_128M_pool_server_config_engines_0_storage_0_1_2_setup_53e1/)
  - [Test Hardware / Functional Hardware Medium / FTEST_aggregation.AggregationChecksum.1-./aggregation/checksum.py:AggregationChecksum.test_aggregationchecksum;run-container-hosts-ior-client_processes-objectclass-SX-transfersize_blocksize-2K-pool-server_config-engines-0-storage-0-1-2-15d2](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/1/testReport/junit/FTEST_aggregation/AggregationChecksum/Test_Hardware___Functional_Hardware_Medium___1___aggregation_checksum_py_AggregationChecksum_test_aggregationchecksum_run_container_hosts_ior_client_processes_objectclass_SX_transfersize_blocksize_2K_pool_server_config_engines_0_storage_0_1_2_15d2/)
  - [Test Hardware / Functional Hardware Medium / FTEST_pool.PoolCreateTests.1-./pool/create.py:PoolCreateTests.test_create_no_space_loop;run-hosts-pool_1-pool_2-server_config-engines-0-storage-0-1-timeouts-9f19](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/1/testReport/junit/FTEST_pool/PoolCreateTests/Test_Hardware___Functional_Hardware_Medium___1___pool_create_py_PoolCreateTests_test_create_no_space_loop_run_hosts_pool_1_pool_2_server_config_engines_0_storage_0_1_timeouts_9f19/)

- Known issue https://daosio.atlassian.net/browse/DAOS-13200:
  - [Test Hardware / Functional Hardware Medium Verbs Provider / FTEST_daos_test.DaosCoreTest.20-./daos_test/suite.py:DaosCoreTest.test_daos_extend_simple;run-agent_config-transport_config-daos_tests-args-daos_test-num_clients-pools_created-scalable_endpoint-stopped_ranks-test_name-dmg-hosts-pool-server_config-engines-0-storage-0-1-timeouts-4c4d](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/1/testReport/junit/FTEST_daos_test/DaosCoreTest/Test_Hardware___Functional_Hardware_Medium_Verbs_Provider___20___daos_test_suite_py_DaosCoreTest_test_daos_extend_simple_run_agent_config_transport_config_daos_tests_args_daos_test_num_clients_pools_created_scalable_endpoint_stopped_ranks_test_name_dmg_hosts_pool_server_config_engines_0_storage_0_1_timeouts_4c4d/)
  - [Test Hardware / Functional Hardware Medium Verbs Provider / FTEST_daos_test.DAOS_Extend_Simple.EXTEND13: read object during extend and extend](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/1/testReport/junit/FTEST_daos_test/DAOS_Extend_Simple/Test_Hardware___Functional_Hardware_Medium_Verbs_Provider___EXTEND13__read_object_during_extend_and_extend/)
  - [Test Hardware / Functional Hardware Medium Verbs Provider / FTEST_daos_test.DAOS_Extend_Simple.EXTEND14: write object during extend and kill](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/1/testReport/junit/FTEST_daos_test/DAOS_Extend_Simple/Test_Hardware___Functional_Hardware_Medium_Verbs_Provider___EXTEND14__write_object_during_extend_and_kill/)
  - [Test Hardware / Functional Hardware Medium Verbs Provider / FTEST_daos_test.DAOS_Extend_Simple.EXTEND15: write object during extend and extend](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/1/testReport/junit/FTEST_daos_test/DAOS_Extend_Simple/Test_Hardware___Functional_Hardware_Medium_Verbs_Provider___EXTEND15__write_object_during_extend_and_extend/)
  - [Test Hardware / Functional Hardware Medium Verbs Provider / FTEST_daos_test.DAOS_Extend_Simple.EXTEND16: extend fail then retry](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/1/testReport/junit/FTEST_daos_test/DAOS_Extend_Simple/Test_Hardware___Functional_Hardware_Medium_Verbs_Provider___EXTEND16__extend_fail_then_retry/)

- Known issue https://daosio.atlassian.net/browse/DAOS-12883:
  - [Test Hardware / Functional Hardware Medium Verbs Provider / FTEST_daos_test.DaosCoreTest.23-./daos_test/suite.py:DaosCoreTest.test_daos_rebuild_ec;run-agent_config-transport_config-daos_tests-args-daos_test-num_clients-pools_created-scalable_endpoint-stopped_ranks-test_name-dmg-hosts-pool-server_config-engines-0-storage-0-1-timeouts-4c4d](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/1/testReport/junit/FTEST_daos_test/DaosCoreTest/Test_Hardware___Functional_Hardware_Medium_Verbs_Provider___23___daos_test_suite_py_DaosCoreTest_test_daos_rebuild_ec_run_agent_config_transport_config_daos_tests_args_daos_test_num_clients_pools_created_scalable_endpoint_stopped_ranks_test_name_dmg_hosts_pool_server_config_engines_0_storage_0_1_timeouts_4c4d/)
  - [Test Hardware / Functional Hardware Medium Verbs Provider / FTEST_daos_test.DAOS_Rebuild_EC.23-./daos_test/suite.py:DaosCoreTest.test_daos_rebuild_ec;run-agent_config-transport_config-daos_tests-args-daos_test-num_clients-pools_created-scalable_endpoint-stopped_ranks-test_name-dmg-hosts-pool-server_config-engines-0-storage-0-1-timeouts-4c4d](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/1/testReport/junit/FTEST_daos_test/DAOS_Rebuild_EC/Test_Hardware___Functional_Hardware_Medium_Verbs_Provider___23___daos_test_suite_py_DaosCoreTest_test_daos_rebuild_ec_run_agent_config_transport_config_daos_tests_args_daos_test_num_clients_pools_created_scalable_endpoint_stopped_ranks_test_name_dmg_hosts_pool_server_config_engines_0_storage_0_1_timeouts_4c4d/)

File: src/tests/ftest/launch.py (hunk 1 above)
Comment (phender) [reply src/tests/ftest/launch.py:1568]: Probably not. I can fix the typo later.

**Comment by phender** [comment]
Test results:

- NEW issue https://daosio.atlassian.net/browse/DAOS-13410
  - [Test Hardware / Functional Hardware Medium / FTEST_daos_test.DAOS_Rebuild.REBUILD22: connect pool during scan for offline rebuild](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_daos_test/DAOS_Rebuild/Test_Hardware___Functional_Hardware_Medium___REBUILD22__connect_pool_during_scan_for_offline_rebuild/)
  - [Test Hardware / Functional Hardware Medium / FTEST_daos_test.DAOS_Rebuild.REBUILD23: connect pool during rebuild for offline rebuild](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_daos_test/DAOS_Rebuild/Test_Hardware___Functional_Hardware_Medium___REBUILD23__connect_pool_during_rebuild_for_offline_rebuild/)
  - [Test Hardware / Functional Hardware Medium / FTEST_daos_test.DAOS_Rebuild.REBUILD24: offline rebuild](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_daos_test/DAOS_Rebuild/Test_Hardware___Functional_Hardware_Medium___REBUILD24__offline_rebuild/)
  - [Test Hardware / Functional Hardware Medium / FTEST_daos_test.DAOS_Rebuild.REBUILD25: rebuild with two failures](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_daos_test/DAOS_Rebuild/Test_Hardware___Functional_Hardware_Medium___REBUILD25__rebuild_with_two_failures/)
  - [Test Hardware / Functional Hardware Medium / FTEST_daos_test.DAOS_Rebuild.REBUILD28: rebuild kill rank during rebuild](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_daos_test/DAOS_Rebuild/Test_Hardware___Functional_Hardware_Medium___REBUILD28__rebuild_kill_rank_during_rebuild/)
  - [Test Hardware / Functional Hardware Medium / FTEST_daos_test.DAOS_Rebuild.REBUILD29: rebuild kill PS leader during rebuild](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_daos_test/DAOS_Rebuild/Test_Hardware___Functional_Hardware_Medium___REBUILD29__rebuild_kill_PS_leader_during_rebuild/)

- Known issue https://daosio.atlassian.net/browse/DAOS-12750 
  - [Test Hardware / Functional Hardware Medium / FTEST_aggregation.AggregationChecksum.1-./aggregation/checksum.py:AggregationChecksum.test_aggregationchecksum;run-container-hosts-ior-client_processes-objectclass-SX-transfersize_blocksize-2K-pool-server_config-control_metadata-engines-0-storage-0-1-2-7a24](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_aggregation/AggregationChecksum/Test_Hardware___Functional_Hardware_Medium___1___aggregation_checksum_py_AggregationChecksum_test_aggregationchecksum_run_container_hosts_ior_client_processes_objectclass_SX_transfersize_blocksize_2K_pool_server_config_control_metadata_engines_0_storage_0_1_2_7a24/)
  - [Test Hardware / Functional Hardware Medium / FTEST_pool.PoolCreateTests.1-./pool/create.py:PoolCreateTests.test_create_no_space_loop;run-hosts-pool_1-pool_2-server_config-control_metadata-engines-0-storage-0-1-timeouts-8076](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_pool/PoolCreateTests/Test_Hardware___Functional_Hardware_Medium___1___pool_create_py_PoolCreateTests_test_create_no_space_loop_run_hosts_pool_1_pool_2_server_config_control_metadata_engines_0_storage_0_1_timeouts_8076/)
  - [Test Hardware / Functional Hardware Large / FTEST_erasurecode.EcodIor.1-./erasurecode/ior_smoke.py:EcodIor.test_ec;run-container-hosts-ior-client_processes-iorflags-objectclass-transfersize_blocksize-1M_128M-pool-server_config-control_metadata-engines-0-storage-0-1-2-setup-2964](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_erasurecode/EcodIor/Test_Hardware___Functional_Hardware_Large___1___erasurecode_ior_smoke_py_EcodIor_test_ec_run_container_hosts_ior_client_processes_iorflags_objectclass_transfersize_blocksize_1M_128M_pool_server_config_control_metadata_engines_0_storage_0_1_2_setup_2964/)
  - [Test Hardware / Functional Hardware Large / FTEST_erasurecode.EcodIor.2-./erasurecode/ior_smoke.py:EcodIor.test_ec;run-container-hosts-ior-client_processes-iorflags-objectclass-transfersize_blocksize-2M_128M-pool-server_config-control_metadata-engines-0-storage-0-1-2-setup-24df](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_erasurecode/EcodIor/Test_Hardware___Functional_Hardware_Large___2___erasurecode_ior_smoke_py_EcodIor_test_ec_run_container_hosts_ior_client_processes_iorflags_objectclass_transfersize_blocksize_2M_128M_pool_server_config_control_metadata_engines_0_storage_0_1_2_setup_24df/)
  - [Test Hardware / Functional Hardware Large / FTEST_erasurecode.EcodMdtest.1-./erasurecode/mdtest_smoke.py:EcodMdtest.test_ec_mdtest_smoke;run-container-hosts-mdtest-client_processes-np_48-diff_flags-flags_ssf-data-zero-mdtest_api-dfs-objectclass-pool-server_config-control_metadata-engines-0-storage-0-1-2-17c3](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_erasurecode/EcodMdtest/Test_Hardware___Functional_Hardware_Large___1___erasurecode_mdtest_smoke_py_EcodMdtest_test_ec_mdtest_smoke_run_container_hosts_mdtest_client_processes_np_48_diff_flags_flags_ssf_data_zero_mdtest_api_dfs_objectclass_pool_server_config_control_metadata_engines_0_storage_0_1_2_17c3/)
  - [Test Hardware / Functional Hardware Large / FTEST_erasurecode.EcodMdtest.2-./erasurecode/mdtest_smoke.py:EcodMdtest.test_ec_mdtest_smoke;run-container-hosts-mdtest-client_processes-np_48-diff_flags-flags_fpp-mdtest_api-dfs-objectclass-pool-server_config-control_metadata-engines-0-storage-0-1-2-93b5](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_erasurecode/EcodMdtest/Test_Hardware___Functional_Hardware_Large___2___erasurecode_mdtest_smoke_py_EcodMdtest_test_ec_mdtest_smoke_run_container_hosts_mdtest_client_processes_np_48_diff_flags_flags_fpp_mdtest_api_dfs_objectclass_pool_server_config_control_metadata_engines_0_storage_0_1_2_93b5/)
  - [Test Hardware / Functional Hardware Large / FTEST_erasurecode.EcodOnlineRebuildMdtest.1-./erasurecode/online_rebuild_mdtest.py:EcodOnlineRebuildMdtest.test_ec_online_rebuild_mdtest;run-container-hosts-servers-6_server-mdtest-client_processes-np_48-mdtest_api-dfs-objectclass-pool-server_config-control_metadata-engines-0-storage-0-1-2-setup-a207](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_erasurecode/EcodOnlineRebuildMdtest/Test_Hardware___Functional_Hardware_Large___1___erasurecode_online_rebuild_mdtest_py_EcodOnlineRebuildMdtest_test_ec_online_rebuild_mdtest_run_container_hosts_servers_6_server_mdtest_client_processes_np_48_mdtest_api_dfs_objectclass_pool_server_config_control_metadata_engines_0_storage_0_1_2_setup_a207/)
  - [Test Hardware / Functional Hardware Large / FTEST_erasurecode.EcodOnlineRebuildMdtest.2-./erasurecode/online_rebuild_mdtest.py:EcodOnlineRebuildMdtest.test_ec_online_rebuild_mdtest;run-container-hosts-servers-8_server-mdtest-client_processes-np_48-mdtest_api-dfs-objectclass-pool-server_config-control_metadata-engines-0-storage-0-1-2-setup-29d5](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_erasurecode/EcodOnlineRebuildMdtest/Test_Hardware___Functional_Hardware_Large___2___erasurecode_online_rebuild_mdtest_py_EcodOnlineRebuildMdtest_test_ec_online_rebuild_mdtest_run_container_hosts_servers_8_server_mdtest_client_processes_np_48_mdtest_api_dfs_objectclass_pool_server_config_control_metadata_engines_0_storage_0_1_2_setup_29d5/)
  - [Test Hardware / Functional Hardware Large / FTEST_erasurecode.EcodOnlineRebuildMdtest.3-./erasurecode/online_rebuild_mdtest.py:EcodOnlineRebuildMdtest.test_ec_online_rebuild_mdtest;run-container-hosts-servers-12_server-mdtest-client_processes-np_48-mdtest_api-dfs-objectclass-pool-server_config-control_metadata-engines-0-storage-0-1-2-setup-c4ad](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12109/13/testReport/junit/FTEST_erasurecode/EcodOnlineRebuildMdtest/Test_Hardware___Functional_Hardware_Large___3___erasurecode_online_rebuild_mdtest_py_EcodOnlineRebuildMdtest_test_ec_online_rebuild_mdtest_run_container_hosts_servers_12_server_mdtest_client_processes_np_48_mdtest_api_dfs_objectclass_pool_server_config_control_metadata_engines_0_storage_0_1_2_setup_c4ad/)

File: src/tests/ftest/util/server_utils_base.py [hunk 2]
```diff
@@ -177,10 +186,21 @@ def engine_params(self):
...
+    def control_metadata(self):
+        """Get the control plane metadata configuration parameters.
+
+        Returns:
+            ControlMetadataParameters: the control plane metadata configuration parameters or None
+                if not defined.
+        """
+        if self.yaml is not None and hasattr(self.yaml, "metadata_params"):
+            return self.yaml.metadata_params
+        return None
```
Comment (daltonbohning) [inline src/tests/ftest/util/server_utils_base.py:203]: Not sure I have a preference, but curious what you think?
```suggestion
        try:
            return self.yaml.metadata_params
        except AttributeError:
            return None
```

**Review by daltonbohning — APPROVED** [review]

**Review by tanabarr — APPROVED** [review]

**Pull request merged**